### PR TITLE
Market

### PR DESCRIPTION
**Pull request by maikelbos0**
I think it's complete but idk

### CODE REVIEW

File: WarOfEmpires/Views/Market/Sell.cshtml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+@{
+    ViewBag.Title = "Sell resources";
+}
+
+<h2>Sell resources</h2>
+
+<p>
+    Send out new caravans to sell your resources on the market, or call back caravans currently on the market. You are able to send off multiple caravans at once,
+    provided you have enough available merchants.
```
Comment (KatarinaSundqvist) [inline WarOfEmpires/Views/Market/Sell.cshtml]: Please include information about the sales tax on this page

File: WarOfEmpires.Domain/Empires/BuildingDefinitionFactory.cs [hunk 2]
```diff
@@ -248,6 +248,22 @@ public static class BuildingDefinitionFactory {
             );
         }
 
+        private static BuildingDefinition GenerateMarket() {
+            var descriptions = new ExpressionGenerator<string>("Your market allows you to send merchants off to sell your excess resources");
+            descriptions.Add(1, SequenceGeneratorFactory.GetGeneratorFunction((value) => $"Your market allows you to send merchants off to sell your excess resources; each merchant caravan can carry {value * 5000} resources"));
```
Comment (KatarinaSundqvist) [inline WarOfEmpires.Domain/Empires/BuildingDefinitionFactory.cs]: Consider whether info on sales tax needs to be added here as well as on the market page

File: WarOfEmpires.Models/Markets/MarketModel.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        public string Wood { get; set; }
+        [RegularExpression("^\\d{0,6}$", ErrorMessage = "Wood price must be a valid number")]
+        public string WoodPrice { get; set; }
+        [DisplayName("Stone")]
+        [RegularExpression("^\\d{0,6}$", ErrorMessage = "Stone must be a valid number")]
+        public string Stone { get; set; }
+        [RegularExpression("^\\d{0,6}$", ErrorMessage = "Stone price must be a valid number")]
+        public string StonePrice { get; set; }
+        [DisplayName("Ore")]
+        [RegularExpression("^\\d{0,6}$", ErrorMessage = "Ore must be a valid number")]
```
Comment (KatarinaSundqvist) [inline WarOfEmpires.Models/Markets/MarketModel.cs]: - Suggest changing the regular expressions for the resources (not the resource prices) to "^\\d{0,8}$" to allow larger quantities to be sold. 

- Suggest changing the validation message for the resources to include a reference to the maximum amount allowed, e.g. "Food must be a valid number and < 100 million" to make it more clear to the user why their transaction was refused

File: WarOfEmpires/Views/Market/Sell.cshtml (hunk 1 above)
Comment (maikelbos0) [reply WarOfEmpires/Views/Market/Sell.cshtml]: Done!

File: WarOfEmpires.Domain/Empires/BuildingDefinitionFactory.cs (hunk 2 above)
Comment (maikelbos0) [reply WarOfEmpires.Domain/Empires/BuildingDefinitionFactory.cs]: Seems out of place here.

File: WarOfEmpires.Models/Markets/MarketModel.cs (hunk 3 above)
Comment (maikelbos0) [reply WarOfEmpires.Models/Markets/MarketModel.cs]: I've increased the max to 8 digits but not changed the message for now

**Pull request merged**